pkg/workloads/docker.go

```diff
@@ -300,46 +298,50 @@ func (d *dockerClient) EnableEventListener() (eventsCh chan<- *EventMessage, err
 	since := time.Now()
 	eo := dTypes.EventsOptions{Since: strconv.FormatInt(since.Unix(), 10)}
 	r, err := d.Events(ctx.Background(), eo)
-	if err != nil {
-		return nil, err
-	}
 
-	go d.listenForDockerEvents(ws, r)
+	go d.listenForDockerEvents(ws, r, err)
 
 	log.Debug("Started to listen for docker events")
 	return nil, nil
 }
 
-func (d *dockerClient) listenForDockerEvents(ws *watcherState, reader io.ReadCloser) {
-	scanner := bufio.NewScanner(reader)
-	for scanner.Scan() {
-		metrics.EventTSContainerd.SetToCurrentTime()
-
-		var e dTypesEvents.Message
-		if err := json.Unmarshal(scanner.Bytes(), &e); err != nil {
-			log.WithError(err).Error("Error while unmarshalling event")
-			time.Sleep(100 * time.Millisecond)
-			continue
-		}
-		if e.ID == "" || e.Type != "container" {
-			continue
-		}
-		log.WithFields(logrus.Fields{
-			"event":               e.Status,
-			logfields.ContainerID: shortContainerID(e.ID),
-		}).Debug("Queueing container event")
-
-		switch e.Status {
-		case "start":
-			ws.enqueueByContainerID(e.ID, &EventMessage{WorkloadID: e.ID, EventType: EventTypeStart})
-		case "die":
-			ws.enqueueByContainerID(e.ID, &EventMessage{WorkloadID: e.ID, EventType: EventTypeDelete})
+func (d *dockerClient) listenForDockerEvents(ws *watcherState, messagesCh <-chan dTypesEvents.Message, errCh <-chan error) {
+	for {
+		select {
+		case err, ok := <-errCh:
+			if !ok {
```

Comment:
This could probably fold the later case in as well, if the code handling is the same. Ie:

```suggestion
			if !ok || err == io.EOF {
```